### PR TITLE
[Minor] Remove serverNode from tags structure when heartbeart timeout

### PR DESCRIPTION
**Pull request by zuston**
### What changes were proposed in this pull request?
Remove serverNode from tags structure when heartbeart timeout

### Why are the changes needed?
Remove serverNode from tags structure when heartbeart timeout

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT

### CODE REVIEW

**Comment by zuston** [comment]
@jerqi Minor fix.

File: coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java [hunk 1]
```diff
@@ -72,14 +72,18 @@ public SimpleClusterManager(CoordinatorConf conf) {
...
       for (ServerNode sn : servers.values()) {
         if (timestamp - sn.getTimestamp() > heartbeatTimeout) {
           LOG.warn("Heartbeat timeout detect, " + sn + " will be removed from node list.");
           deleteIds.add(sn.getId());
+          for (Set<ServerNode> nodesWithTag : tagToNodes.values()) {
+            nodesWithTag.remove(sn);
+          }
         }
       }
       for (String serverId : deleteIds) {
```
Comment (jerqi) [inline coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java:89]: How about processing the delete nodesWithTag logic here? And cc @colinma

File: coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java [hunk 2]
```diff
@@ -72,14 +72,18 @@ public SimpleClusterManager(CoordinatorConf conf) {
     }
   }
 
-  private void nodesCheck() {
+  @VisibleForTesting
+  protected void nodesCheck() {
```
Comment (jerqi) [inline coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java:76]: I prefer `default access modifier` than `protected access modifier`. Test code can access the same package source code.

File: coordinator/src/main/java/com/tencent/rss/coordinator/ServerNode.java [hunk 3]
```diff
@@ -115,6 +115,13 @@ public String toString() {
         + ", healthy[" + isHealthy + "]";
   }
 
+  /**
+   * Only for test case
+   */
+  public void setTimestamp(long timestamp) {
```
Comment (jerqi) [inline coordinator/src/main/java/com/tencent/rss/coordinator/ServerNode.java:121]: If we just use in the test, we can use `default access modifier`.

File: coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java (hunk 2 above)
Comment (jerqi) [inline coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java:76]: I prefer use `default` accessor to `protected`. Some spark code use `default` accessor to provide

**Comment by zuston** [comment]
Updated @jerqi

File: coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java [hunk 4]
```diff
@@ -72,14 +72,17 @@ public SimpleClusterManager(CoordinatorConf conf) {
...
       for (ServerNode sn : servers.values()) {
         if (timestamp - sn.getTimestamp() > heartbeatTimeout) {
           LOG.warn("Heartbeat timeout detect, " + sn + " will be removed from node list.");
           deleteIds.add(sn.getId());
+          for (Set<ServerNode> nodesWithTag : tagToNodes.values()) {
+            nodesWithTag.remove(sn);
+          }
         }
       }
       for (String serverId : deleteIds) {
```
Comment (jerqi) [inline coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java:88]: for (String serverId : deleteIds) {
      ServerNode node = servers.remove(serverId);
      if (node != null) {
           for (Set<ServerNode> nodesWithTag : tagToNodes.values()) {
                 nodesWithTag.remove(sn);
          }
      }
}
In the such way, we can put two places deletion logic together.

File: coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java (hunk 1 above)
Comment (zuston) [reply coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java:88]: Sorry, it's my fault. Put together with `deleteIds`'s logic will be better.

File: coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java (hunk 4 above)
Comment (zuston) [reply coordinator/src/main/java/com/tencent/rss/coordinator/SimpleClusterManager.java:88]: Yep. It looks better.

**Comment by jerqi** [comment]
LGTM, @colinmjj Do you have another suggestion?

**Review by colinmjj — APPROVED** [review]
LGTM

**Pull request merged**